torchgeometry/image/laplacian.py

```diff
@@ -0,0 +1,169 @@
+from typing import Tuple
+
+import math
+import torch
+import torch.nn as nn
+from torch.nn.functional import conv2d
+
+
+def laplacian(window_size):
+    r"""One could also use the Laplacian of Gaussian formula
+        to design the filter.
+    """
+
+    laplacian = torch.ones(window_size)
+    laplacian[int(math.floor(window_size / 2))] = 1 - window_size
+    return laplacian
+
+
+def get_laplacian_kernel(ksize: int):
+    r"""Function that returns the coefficients of a 1D Laplacian filter
+
+    Args:
+        ksize (int): filter size. It should be odd and positive.
+
+    Returns:
+        Tensor (float): 1D tensor with laplacian filter coefficients.
+
+    Shape:
+        - Output: math:`(ksize, 0)`
+
+    Examples::
+        >>> tgm.image.get_laplacian_kernel(3)
+        tensor([ 1., -2.,  1.])
+
+        >>> tgm.image.get_laplacian_kernel(5)
+        tensor([ 1.,  1., -4.,  1.,  1.])
+
+    """
+    if not isinstance(ksize, int) or ksize % 2 == 0 or ksize <= 0:
+        raise TypeError("ksize must be an odd positive integer. Got {}"
+                        .format(ksize))
+    window_1d: torch.Tensor = laplacian(ksize)
+    return window_1d
+
+
+def get_laplacian_kernel2d(ksize: int) -> torch.Tensor:
+    r"""Function that returns Gaussian filter matrix coefficients.
+
+    Args:
+        ksize (int): filter size should be odd.
+
+    Returns:
+        Tensor: 2D tensor with laplacian filter matrix coefficients.
+
+    Shape:
+        - Output: :math:`(ksize, ksize)`
+
+    Examples::
+
+        >>> tgm.image.get_laplacian_kernel2d(3)
+        tensor([[ 1.,  1.,  1.],
+        [ 1., -8.,  1.],
+        [ 1.,  1.,  1.]])
+
+        >>> tgm.image.get_laplacian_kernel2d(5)
+        tensor([[  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1., -24.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.],
+        [  1.,   1.,   1.,   1.,   1.]])
+
+    """
+    if not isinstance(ksize, int) or ksize % 2 == 0 or ksize <= 0:
+        raise TypeError("ksize must be a tuple of length two. Got {}"
+                        .format(ksize))
+
```

Comment:
are you sure is this correct?
https://github.com/opencv/opencv/blob/7fb70e170154d064ef12d8fec61c0ae70812ce3d/modules/imgproc/src/deriv.cpp#L797

Reply:
It looks like OpenCV supports for kernel sizes: 3,5, and 7  according to:

https://docs.opencv.org/2.4/modules/imgproc/doc/filtering.html?highlight=laplacian#void%20getDerivKernels(OutputArray%20kx,%20OutputArray%20ky,%20int%20dx,%20int%20dy,%20int%20ksize,%20bool%20normalize,%20int%20ktype)

The kernel coefficients you referred to in this link are for a 3 by 3 filter. The formula which I wrote is much more generic one as described here:

https://homepages.inf.ed.ac.uk/rbf/HIPR2/log.htm